### ISSUE TITLE
ActionController::Parameters issue with nested parameters

### ISSUE THREAD

**Issue by danielpuglisi** [chrisfrank/rack-reducer#13]
Hi @chrisfrank 

I just stumbled across an issue with Rails's `ActionController::Parameters` in filters with nested parameters. Example:

```
->(date:) { between(date[:from], date[:to]) }
```

This results in `nil` values for `date[:from]` and `date[:to]`, because Rails is converting `ActionController::Parameters` to `HashWithIndifferentAccess` objects when calling `#to_unsafe_h` in https://github.com/chrisfrank/rack-reducer/blob/master/lib/rack/reducer.rb#L60. Which doesn't trigger the `Hash#symbolize_keys` from the refinements but the default implementation, causing the nested parameters to be converted back to strings.

```
pry(main)> ActionController::Parameters.new({ date: { 'from' => '2019-01-01' } }).to_unsafe_h.symbolize_keys
=> {:date=>{"from"=>"2019-01-01"}}
```

Not sure how to solve this yet, but I'm looking for a solution.

**Comment by chrisfrank** [comment]
Thanks for the detailed report, @danielpuglisi, and for opening a PR with a promising fix. I'll have time to review it over the weekend, and hopefully we can merge then or early next week!

**Comment by chrisfrank** [comment]
Fixed by @danielpuglisi in #14 🚀
